In [295]:
%matplotlib inline
import warnings
warnings.filterwarnings('ignore')
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import os
from sklearn import linear_model, datasets, tree
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split
from sklearn.svm import SVC 
from sklearn.datasets import make_blobs
from sklearn.model_selection import GridSearchCV
import seaborn as sns
from pandas.plotting import scatter_matrix
from sklearn.tree import DecisionTreeClassifier
from matplotlib import pyplot
from sklearn import metrics

# Read the CSV and Perform Basic Data Cleaning

In [296]:
df = pd.read_csv("exoplanet_data.csv")
df.head()

koi_disposition  koi_fpflag_nt  koi_fpflag_ss  koi_fpflag_co  koi_fpflag_ec  \
0       CONFIRMED              0              0              0              0   
1  FALSE POSITIVE              0              1              0              0   
2  FALSE POSITIVE              0              1              0              0   
3       CONFIRMED              0              0              0              0   
4       CONFIRMED              0              0              0              0   

   koi_period  koi_period_err1  koi_period_err2  koi_time0bk  \
0   54.418383     2.479000e-04    -2.479000e-04   162.513840   
1   19.899140     1.490000e-05    -1.490000e-05   175.850252   
2    1.736952     2.630000e-07    -2.630000e-07   170.307565   
3    2.525592     3.760000e-06    -3.760000e-06   171.595550   
4    4.134435     1.050000e-05    -1.050000e-05   172.979370   

   koi_time0bk_err1  ...  koi_steff_err2  koi_slogg  koi_slogg_err1  \
0          0.003520  ...             -81      4.467           0.064   
1          0.000581  ...            -176      4.544           0.044   
2          0.000115  ...            -174      4.564           0.053   
3          0.001130  ...            -211      4.438           0.070   
4          0.001900  ...            -232      4.486           0.054   

   koi_slogg_err2  koi_srad  koi_srad_err1  koi_srad_err2         ra  \
0          -0.096     0.927          0.105         -0.061  291.93423   
1          -0.176     0.868          0.233         -0.078  297.00482   
2          -0.168     0.791          0.201         -0.067  285.53461   
3          -0.210     1.046          0.334         -0.133  288.75488   
4          -0.229     0.972          0.315         -0.105  296.28613   

         dec  koi_kepmag  
0  48.141651      15.347  
1  48.134129      15.436  
2  48.285210      15.597  
3  48.226200      15.509  
4  48.224670      15.714  

[5 rows x 41 columns]

In [297]:
df.columns

Index(['koi_disposition', 'koi_fpflag_nt', 'koi_fpflag_ss', 'koi_fpflag_co',
       'koi_fpflag_ec', 'koi_period', 'koi_period_err1', 'koi_period_err2',
       'koi_time0bk', 'koi_time0bk_err1', 'koi_time0bk_err2', 'koi_impact',
       'koi_impact_err1', 'koi_impact_err2', 'koi_duration',
       'koi_duration_err1', 'koi_duration_err2', 'koi_depth', 'koi_depth_err1',
       'koi_depth_err2', 'koi_prad', 'koi_prad_err1', 'koi_prad_err2',
       'koi_teq', 'koi_insol', 'koi_insol_err1', 'koi_insol_err2',
       'koi_model_snr', 'koi_tce_plnt_num', 'koi_steff', 'koi_steff_err1',
       'koi_steff_err2', 'koi_slogg', 'koi_slogg_err1', 'koi_slogg_err2',
       'koi_srad', 'koi_srad_err1', 'koi_srad_err2', 'ra', 'dec',
       'koi_kepmag'],
      dtype='object')

In [298]:
df.shape

(6991, 41)

In [299]:
df.describe().transpose()

count          mean            std           min  \
koi_fpflag_nt      6991.0      0.157059       0.363882  0.000000e+00   
koi_fpflag_ss      6991.0      0.244743       0.429966  0.000000e+00   
koi_fpflag_co      6991.0      0.202975       0.402243  0.000000e+00   
koi_fpflag_ec      6991.0      0.125018       0.330763  0.000000e+00   
koi_period         6991.0     56.191248     117.570962  2.598197e-01   
koi_period_err1    6991.0      0.001851       0.007185  1.100000e-08   
koi_period_err2    6991.0     -0.001851       0.007185 -1.568000e-01   
koi_time0bk        6991.0    164.488820      67.020475  1.205159e+02   
koi_time0bk_err1   6991.0      0.009340       0.021989  8.700000e-06   
koi_time0bk_err2   6991.0     -0.009340       0.021989 -5.690000e-01   
koi_impact         6991.0      0.730631       3.346956  0.000000e+00   
koi_impact_err1    6991.0      1.945368       9.401850  0.000000e+00   
koi_impact_err2    6991.0     -0.337243       1.349036 -5.932000e+01   
koi_duration       6991.0      5.549368       6.425205  1.670000e-01   
koi_duration_err1  6991.0      0.320715       0.665352  0.000000e+00   
koi_duration_err2  6991.0     -0.320715       0.665352 -2.020000e+01   
koi_depth          6991.0  23911.900615   82412.402875  7.500000e+00   
koi_depth_err1     6991.0    130.789186    4689.233030  0.000000e+00   
koi_depth_err2     6991.0   -130.789186    4689.233030 -3.886000e+05   
koi_prad           6991.0    121.194193    3519.737626  1.400000e-01   
koi_prad_err1      6991.0     19.483454     436.752064  0.000000e+00   
koi_prad_err2      6991.0    -40.352509    1366.939375 -7.718000e+04   
koi_teq            6991.0   1095.152768     856.234809  9.200000e+01   
koi_insol          6991.0   8263.768510  177525.467655  2.000000e-02   
koi_insol_err1     6991.0   3599.917979   56072.701854  0.000000e+00   
koi_insol_err2     6991.0  -4302.621498   99248.481985 -5.600031e+06   
koi_model_snr      6991.0    276.038764     824.788481  1.600000e+00   
koi_tce_plnt_num   6991.0      1.233729       0.650089  1.000000e+00   
koi_steff          6991.0   5700.682449     802.060405  2.661000e+03   
koi_steff_err1     6991.0    143.898155      46.566511  0.000000e+00   
koi_steff_err2     6991.0   -161.206980      71.448481 -1.733000e+03   
koi_slogg          6991.0      4.305049       0.439238  4.700000e-02   
koi_slogg_err1     6991.0      0.121091       0.132048  0.000000e+00   
koi_slogg_err2     6991.0     -0.140480       0.081990 -1.007000e+00   
koi_srad           6991.0      1.740749       5.903415  1.090000e-01   
koi_srad_err1      6991.0      0.352710       0.839017  0.000000e+00   
koi_srad_err2      6991.0     -0.388568       1.907797 -1.038250e+02   
ra                 6991.0    292.082406       4.762908  2.798561e+02   
dec                6991.0     43.812143       3.606167  3.657738e+01   
koi_kepmag         6991.0     14.271508       1.350802  6.966000e+00   

                           25%          50%          75%           max  
koi_fpflag_nt         0.000000     0.000000     0.000000  1.000000e+00  
koi_fpflag_ss         0.000000     0.000000     0.000000  1.000000e+00  
koi_fpflag_co         0.000000     0.000000     0.000000  1.000000e+00  
koi_fpflag_ec         0.000000     0.000000     0.000000  1.000000e+00  
koi_period            2.620126     8.947426    34.282605  1.071233e+03  
koi_period_err1       0.000005     0.000033     0.000240  1.568000e-01  
koi_period_err2      -0.000240    -0.000033    -0.000005 -1.100000e-08  
koi_time0bk         132.683917   136.739230   169.937005  1.472522e+03  
koi_time0bk_err1      0.001145     0.003990     0.010000  5.690000e-01  
koi_time0bk_err2     -0.010000    -0.003990    -0.001145 -8.700000e-06  
koi_impact            0.198000     0.539000     0.891000  1.008060e+02  
koi_impact_err1       0.039900     0.189800     0.379000  8.554000e+01  
koi_impact_err2      -0.446000    -0.204000    -0.044000  0.000000e+00  
koi_duration          2.446000     3.793000  

In [300]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6991 entries, 0 to 6990
Data columns (total 41 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   koi_disposition    6991 non-null   object 
 1   koi_fpflag_nt      6991 non-null   int64  
 2   koi_fpflag_ss      6991 non-null   int64  
 3   koi_fpflag_co      6991 non-null   int64  
 4   koi_fpflag_ec      6991 non-null   int64  
 5   koi_period         6991 non-null   float64
 6   koi_period_err1    6991 non-null   float64
 7   koi_period_err2    6991 non-null   float64
 8   koi_time0bk        6991 non-null   float64
 9   koi_time0bk_err1   6991 non-null   float64
 10  koi_time0bk_err2   6991 non-null   float64
 11  koi_impact         6991 non-null   float64
 12  koi_impact_err1    6991 non-null   float64
 13  koi_impact_err2    6991 non-null   float64
 14  koi_duration       6991 non-null   float64
 15  koi_duration_err1  6991 non-null   float64
 16  koi_duration_err2  6991 

In [301]:
#no null values
df.isnull().sum(axis = 0)

koi_disposition      0
koi_fpflag_nt        0
koi_fpflag_ss        0
koi_fpflag_co        0
koi_fpflag_ec        0
koi_period           0
koi_period_err1      0
koi_period_err2      0
koi_time0bk          0
koi_time0bk_err1     0
koi_time0bk_err2     0
koi_impact           0
koi_impact_err1      0
koi_impact_err2      0
koi_duration         0
koi_duration_err1    0
koi_duration_err2    0
koi_depth            0
koi_depth_err1       0
koi_depth_err2       0
koi_prad             0
koi_prad_err1        0
koi_prad_err2        0
koi_teq              0
koi_insol            0
koi_insol_err1       0
koi_insol_err2       0
koi_model_snr        0
koi_tce_plnt_num     0
koi_steff            0
koi_steff_err1       0
koi_steff_err2       0
koi_slogg            0
koi_slogg_err1       0
koi_slogg_err2       0
koi_srad             0
koi_srad_err1        0
koi_srad_err2        0
ra                   0
dec                  0
koi_kepmag           0
dtype: int64

# Select your features (columns)

In [302]:
# Set features. This will also be used as your x values.
selected_features = df[['koi_disposition', 'koi_fpflag_nt', 'koi_fpflag_ss', 'koi_fpflag_co',
       'koi_fpflag_ec', 'koi_period', 'koi_period_err1', 'koi_period_err2',
       'koi_time0bk', 'koi_time0bk_err1', 'koi_time0bk_err2', 'koi_impact',
       'koi_impact_err1', 'koi_impact_err2', 'koi_duration',
       'koi_duration_err1', 'koi_duration_err2', 'koi_depth', 'koi_depth_err1',
       'koi_depth_err2', 'koi_prad', 'koi_prad_err1', 'koi_prad_err2',
       'koi_teq', 'koi_insol', 'koi_insol_err1', 'koi_insol_err2',
       'koi_model_snr', 'koi_tce_plnt_num', 'koi_steff', 'koi_steff_err1',
       'koi_steff_err2', 'koi_slogg', 'koi_slogg_err1', 'koi_slogg_err2',
       'koi_srad', 'koi_srad_err1', 'koi_srad_err2', 'ra', 'dec',
       'koi_kepmag']]


In [303]:
# Set features. This will also be used as your x values.
selected_names = df[[ 'koi_fpflag_nt', 'koi_fpflag_ss', 'koi_fpflag_co',
       'koi_fpflag_ec', 'koi_period', 'koi_period_err1', 'koi_period_err2',
       'koi_time0bk', 'koi_time0bk_err1', 'koi_time0bk_err2', 'koi_impact',
       'koi_impact_err1', 'koi_impact_err2', 'koi_duration',
       'koi_duration_err1', 'koi_duration_err2', 'koi_depth', 'koi_depth_err1',
       'koi_depth_err2', 'koi_prad', 'koi_prad_err1', 'koi_prad_err2',
       'koi_teq', 'koi_insol', 'koi_insol_err1', 'koi_insol_err2',
       'koi_model_snr', 'koi_tce_plnt_num', 'koi_steff', 'koi_steff_err1',
       'koi_steff_err2', 'koi_slogg', 'koi_slogg_err1', 'koi_slogg_err2',
       'koi_srad', 'koi_srad_err1', 'koi_srad_err2', 'ra', 'dec',
       'koi_kepmag']]


In [304]:
selected_features.head()

koi_disposition  koi_fpflag_nt  koi_fpflag_ss  koi_fpflag_co  koi_fpflag_ec  \
0       CONFIRMED              0              0              0              0   
1  FALSE POSITIVE              0              1              0              0   
2  FALSE POSITIVE              0              1              0              0   
3       CONFIRMED              0              0              0              0   
4       CONFIRMED              0              0              0              0   

   koi_period  koi_period_err1  koi_period_err2  koi_time0bk  \
0   54.418383     2.479000e-04    -2.479000e-04   162.513840   
1   19.899140     1.490000e-05    -1.490000e-05   175.850252   
2    1.736952     2.630000e-07    -2.630000e-07   170.307565   
3    2.525592     3.760000e-06    -3.760000e-06   171.595550   
4    4.134435     1.050000e-05    -1.050000e-05   172.979370   

   koi_time0bk_err1  ...  koi_steff_err2  koi_slogg  koi_slogg_err1  \
0          0.003520  ...             -81      4.467           0.064   
1          0.000581  ...            -176      4.544           0.044   
2          0.000115  ...            -174      4.564           0.053   
3          0.001130  ...            -211      4.438           0.070   
4          0.001900  ...            -232      4.486           0.054   

   koi_slogg_err2  koi_srad  koi_srad_err1  koi_srad_err2         ra  \
0          -0.096     0.927          0.105         -0.061  291.93423   
1          -0.176     0.868          0.233         -0.078  297.00482   
2          -0.168     0.791          0.201         -0.067  285.53461   
3          -0.210     1.046          0.334         -0.133  288.75488   
4          -0.229     0.972          0.315         -0.105  296.28613   

         dec  koi_kepmag  
0  48.141651      15.347  
1  48.134129      15.436  
2  48.285210      15.597  
3  48.226200      15.509  
4  48.224670      15.714  

[5 rows x 41 columns]

In [305]:
for feature in selected_features.columns: # Loop through all columns in the dataframe
    if selected_features[feature].dtype == 'object': # Only apply for columns with categorical strings
        selected_features[feature] = pd.Categorical(selected_features[feature])# Replace strings with an integer
selected_features.head(10)

koi_disposition  koi_fpflag_nt  koi_fpflag_ss  koi_fpflag_co  koi_fpflag_ec  \
0       CONFIRMED              0              0              0              0   
1  FALSE POSITIVE              0              1              0              0   
2  FALSE POSITIVE              0              1              0              0   
3       CONFIRMED              0              0              0              0   
4       CONFIRMED              0              0              0              0   
5       CONFIRMED              0              0              0              0   
6       CONFIRMED              0              0              0              0   
7       CONFIRMED              0              0              0              0   
8       CONFIRMED              0              1              0              0   
9       CONFIRMED              0              0              0              0   

   koi_period  koi_period_err1  koi_period_err2  koi_time0bk  \
0   54.418383     2.479000e-04    -2.479000e-04   162.513840   
1   19.899140     1.490000e-05    -1.490000e-05   175.850252   
2    1.736952     2.630000e-07    -2.630000e-07   170.307565   
3    2.525592     3.760000e-06    -3.760000e-06   171.595550   
4    4.134435     1.050000e-05    -1.050000e-05   172.979370   
5    2.566589     1.780000e-05    -1.780000e-05   179.554370   
6   16.068647     1.090000e-05    -1.090000e-05   173.621937   
7    2.470613     2.700000e-08    -2.700000e-08   122.763305   
8    2.204735     4.300000e-08    -4.300000e-08   121.358542   
9    3.522498     1.980000e-07    -1.980000e-07   121.119423   

   koi_time0bk_err1  ...  koi_steff_err2  koi_slogg  koi_slogg_err1  \
0          0.003520  ...             -81      4.467           0.064   
1          0.000581  ...            -176      4.544           0.044   
2          0.000115  ...            -174      4.564           0.053   
3          0.001130  ...            -211      4.438           0.070   
4          0.001900  ...            -232      4.486           0.054   
5          0.004610  ...            -232      4.486           0.054   
6          0.000517  ...             -83      4.485           0.083   
7          0.000009  ...             -78      4.457           0.024   
8          0.000016  ...             -89      4.019           0.033   
9          0.000047  ...            -137      4.169           0.055   

   koi_slogg_err2  koi_srad  koi_srad_err1  koi_srad_err2         ra  \
0          -0.096     0.927          0.105         -0.061  291.93423   
1          -0.176     0.868          0.233         -0.078  297.00482   
2          -0.168     0.791          0.201         -0.067  285.53461   
3          -0.210     1.046          0.334         -0.133  288.75488   
4          -0.229     0.972          0.315         -0.105  296.28613   
5          -0.229     0.972          0.315         -0.105  296.28613   
6          -0.028     0.848          0.033         -0.072  286.99948   
7          -0.024     0.964          0.038         -0.038  286.80847   
8          -0.027     1.952          0.099         -0.110  292.24728   
9          -0.045     1.451          0.110         -0.110  281.28812   

         dec  koi_kepmag  
0  48.141651      15.347  
1  48.134129      15.436  
2  48.285210      15.597  
3  48.226200      15.509  
4  48.224670      15.714  
5  48.224670      15.714  
6  48.375790      15.841  
7  49.316399      11.338  
8  47.969521      10.463  
9  42.451080      13.563  

[10 rows x 41 columns]

In [306]:
print(selected_features.koi_disposition.value_counts())

FALSE POSITIVE    3504
CONFIRMED         1800
CANDIDATE         1687
Name: koi_disposition, dtype: int64


In [307]:
replaceStruct = {
                "koi_disposition":     {"CONFIRMED": 1, "FALSE POSITIVE": 2 ,"CANDIDATE": 3}
                    }
df_features=selected_features.replace(replaceStruct)
df_features.head(10)

koi_disposition  koi_fpflag_nt  koi_fpflag_ss  koi_fpflag_co  \
0                1              0              0              0   
1                2              0              1              0   
2                2              0              1              0   
3                1              0              0              0   
4                1              0              0              0   
5                1              0              0              0   
6                1              0              0              0   
7                1              0              0              0   
8                1              0              1              0   
9                1              0              0              0   

   koi_fpflag_ec  koi_period  koi_period_err1  koi_period_err2  koi_time0bk  \
0              0   54.418383     2.479000e-04    -2.479000e-04   162.513840   
1              0   19.899140     1.490000e-05    -1.490000e-05   175.850252   
2              0    1.736952     2.630000e-07    -2.630000e-07   170.307565   
3              0    2.525592     3.760000e-06    -3.760000e-06   171.595550   
4              0    4.134435     1.050000e-05    -1.050000e-05   172.979370   
5              0    2.566589     1.780000e-05    -1.780000e-05   179.554370   
6              0   16.068647     1.090000e-05    -1.090000e-05   173.621937   
7              0    2.470613     2.700000e-08    -2.700000e-08   122.763305   
8              0    2.204735     4.300000e-08    -4.300000e-08   121.358542   
9              0    3.522498     1.980000e-07    -1.980000e-07   121.119423   

   koi_time0bk_err1  ...  koi_steff_err2  koi_slogg  koi_slogg_err1  \
0          0.003520  ...             -81      4.467           0.064   
1          0.000581  ...            -176      4.544           0.044   
2          0.000115  ...            -174      4.564           0.053   
3          0.001130  ...            -211      4.438           0.070   
4          0.001900  ...            -232      4.486           0.054   
5          0.004610  ...            -232      4.486           0.054   
6          0.000517  ...             -83      4.485           0.083   
7          0.000009  ...             -78      4.457           0.024   
8          0.000016  ...             -89      4.019           0.033   
9          0.000047  ...            -137      4.169           0.055   

   koi_slogg_err2  koi_srad  koi_srad_err1  koi_srad_err2         ra  \
0          -0.096     0.927          0.105         -0.061  291.93423   
1          -0.176     0.868          0.233         -0.078  297.00482   
2          -0.168     0.791          0.201         -0.067  285.53461   
3          -0.210     1.046          0.334         -0.133  288.75488   
4          -0.229     0.972          0.315         -0.105  296.28613   
5          -0.229     0.972          0.315         -0.105  296.28613   
6          -0.028     0.848          0.033         -0.072  286.99948   
7          -0.024     0.964          0.038         -0.038  286.80847   
8          -0.027     1.952          0.099         -0.110  292.24728   
9          -0.045     1.451          0.110         -0.110  281.28812   

         dec  koi_kepmag  
0  48.141651      15.347  
1  48.134129      15.436  
2  48.285210      15.597  
3  48.226200      15.509  
4  48.224670      15.714  
5  48.224670      15.714  
6  48.375790      15.841  
7  49.316399      11.338  
8  47.969521      10.463  
9  42.451080      13.563  

[10 rows x 41 columns]

# Create a Train Test Split

Use `koi_disposition` for the y values

In [308]:
y = df_features['koi_disposition']
print(y.shape)


(6991,)


In [309]:
X = df_features.drop("koi_disposition" , axis=1)
y = df_features.pop("koi_disposition")

In [310]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42)

# Pre-processing & Train the Model



In [311]:
from sklearn.preprocessing import MinMaxScaler
X_scaler = MinMaxScaler().fit(X_train)

X_train_scaled = X_scaler.transform(X_train)
X_test_scaled = X_scaler.transform(X_test)

In [312]:
dTree = DecisionTreeClassifier(criterion = 'gini', random_state=42)
dTree.fit(X_train_scaled, y_train)


DecisionTreeClassifier(random_state=42)

In [313]:
importances = dTree.feature_importances_
importances

array([0.1949848 , 0.17331824, 0.18322372, 0.02983544, 0.01008194,
       0.00462449, 0.00372294, 0.01608003, 0.01418874, 0.00349525,
       0.02414054, 0.00848183, 0.00455197, 0.01141872, 0.00895003,
       0.00429295, 0.00487138, 0.00448746, 0.0117903 , 0.01176515,
       0.00720973, 0.00463016, 0.00410508, 0.006598  , 0.00860821,
       0.00281821, 0.13490072, 0.00249952, 0.00746632, 0.0106586 ,
       0.00907781, 0.00240092, 0.00648349, 0.00828513, 0.00181647,
       0.01504706, 0.00492478, 0.00876591, 0.01811001, 0.00728796])

In [314]:
print("Accuracy on training set : ",dTree.score(X_train_scaled, y_train))
print("Accuracy on test set : ",dTree.score(X_test_scaled, y_test))


Accuracy on training set :  1.0
Accuracy on test set :  0.8495423340961098


In [315]:
from sklearn.ensemble import RandomForestClassifier
rf = RandomForestClassifier(n_estimators=200)
rf = rf.fit(X_train_scaled, y_train)
rf.score(X_test_scaled, y_test)

0.902745995423341

In [316]:
sorted(zip(rf.feature_importances_, selected_names), reverse=True)

[(0.11169736633717402, 'koi_fpflag_co'),
 (0.10065369616587266, 'koi_fpflag_nt'),
 (0.06718339063753971, 'koi_fpflag_ss'),
 (0.05743506720721577, 'koi_prad'),
 (0.05445051891682683, 'koi_model_snr'),
 (0.035944082769166344, 'koi_duration_err1'),
 (0.03417663786055775, 'koi_duration_err2'),
 (0.03214242860591074, 'koi_prad_err1'),
 (0.031708720985621074, 'koi_fpflag_ec'),
 (0.03040672168130017, 'koi_steff_err1'),
 (0.02880736791013284, 'koi_prad_err2'),
 (0.02646497551439691, 'koi_steff_err2'),
 (0.024688429482126354, 'koi_time0bk_err2'),
 (0.023606728881396463, 'koi_time0bk_err1'),
 (0.02182604609361285, 'koi_duration'),
 (0.021805445389031338, 'koi_period'),
 (0.01901583805406106, 'koi_impact'),
 (0.017524602371551585, 'koi_depth'),
 (0.017479566051663262, 'koi_period_err1'),
 (0.017149673131624243, 'koi_insol_err1'),
 (0.016388912993541595, 'koi_period_err2'),
 (0.01604597766999233, 'koi_teq'),
 (0.014708206841016983, 'koi_depth_err1'),
 (0.014412456835422381, 'koi_insol'),
 (0.01318

# Save the Model

In [317]:
# save your model by updating "your_name" with your name
# and "your_model" with your model variable
# be sure to turn this in to BCS
# if joblib fails to import, try running the command to install in terminal/git-bash
import joblib
filename = 'DTree_RF.sav'
joblib.dump(model, filename)

['DTree_RF.sav']

##### Conclusions: RandomForest give a higher score compared to Logistic Regresssion model.  90% for RandomForest vs 87% for Logistic Regression model. 